pkg/templates/aws/edges/cloudfront_distribution-load_balancer.yaml

```diff
@@ -0,0 +1,88 @@
+source: aws:cloudfront_distribution
+target: aws:load_balancer
+always_process: true
+operational_rules:
+  - configuration_rules:
+      - resource: '{{ .Source }}'
+        configuration:
+          field: Origins
+          value:
+            - CustomOriginConfig:
+                HttpPort: 80
+                HttpsPort: 443
+                OriginProtocolPolicy: http-only
+                OriginSslProtocols:
+                  - TLSv1.2
+                  - TLSv1
+                  - SSLv3
+                  - TLSv1.1
+              DomainName: '{{ .Target }}#DnsName'
+              OriginId: '{{ .Target.Name }}'
+      - resource: '{{ .Source }}'
+        configuration:
+          field: CacheBehaviors
+          value: |
+            [   
+            {{ $targetOriginId := .Target.Name }}
+            {{ $routes := makeSlice }}
+            {{ $rules := allDownstream "aws:load_balancer_listener_rule" .Target }}
+            {{ range $i, $rule := $rules }}
+              {{ range $j, $condition := (fieldValue "Conditions" $rule) }}
+                {{range $k, $pathPattern := $condition.PathPattern.Values }}
+                  {{- if not (sliceContains $routes $pathPattern) }}
+                    {{ $routes = appendSlice $routes $pathPattern }}
+                  {{- end }}
+                {{- end }}
+              {{ end }}
+            {{ end }}
+            {{ range $i, $route := $routes }}
+              {{ $methods := makeSlice }}
+              {{ range $i, $rule := $rules }}
+                {{ range $j, $condition := (fieldValue "Conditions" $rule) }}
+                  {{- if (sliceContains $condition.PathPattern.Values $route) }}
+                    {{ range $j, $method := $condition.HttpRequestMethod.Values }}
+                      {{ $methods = appendSlice $methods $method }}
+                    {{- end }}
+                  {{- end }}
+                {{- end }}
+              {{ end }}
+              {
+                "AllowedMethods": [
+                    {{- if or (sliceContains $methods "PATCH") (sliceContains $methods "POST") (sliceContains $methods "PUT") (sliceContains $methods "DELETE") }}
```

Comment:
Is there a reason this wouldn't be just the `$methods`?

Reply:
cloudfront has to have them be set in certain groups, not really a free formed list